src/Features/CSharp/Portable/Diagnostics/Analyzers/TypeSyntaxSimplifierWalker.cs

```diff
@@ -4,20 +4,54 @@
 
 using System;
 using System.Collections.Generic;
+using System.Collections.Immutable;
 using System.Threading;
 using Microsoft.CodeAnalysis.CSharp.Extensions;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
 using Microsoft.CodeAnalysis.Options;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.CSharp.Diagnostics.SimplifyTypeNames
 {
     internal class TypeSyntaxSimplifierWalker : CSharpSyntaxWalker
     {
+        private static readonly ImmutableHashSet<string> s_emptyAliasedNames = ImmutableHashSet.Create<string>(StringComparer.Ordinal);
+
+        /// <summary>
+        /// This set contains the full names of types that have equivalent predefined names in the language.
+        /// </summary>
+        private static readonly ImmutableHashSet<string> s_predefinedTypeMetadataNames =
```

Comment:
note: immutablehashsets aren't the fastest.  consider just a normal hashset.

Reply:
It wasn't showing up as observable, but I'll be watching to see if this ever changes.

Reply:
sgtm